src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py

```diff
@@ -68,7 +68,8 @@ def load_arguments(self, _):
     with self.argument_context('ad sp create-for-rbac') as c:
         c.argument('scopes', nargs='+')
         c.argument('role', completer=get_role_definition_name_completion_list)
-        c.argument('skip_assignment', arg_type=get_three_state_flag(), help='do not create default assignment')
+        c.argument('skip_assignment', arg_type=get_three_state_flag(),
+                   help='do not create the default assignment which lets the service principal access resources under current subscription')
```

Comment:
The wording here is a little confusing to me. I would suggest:
`Skip creating the default assignment. This allows the service principal to access resources under the current subscription.`  

Also, what if they user supplied `--subscription`?

Reply:
The `--subscription` flag has not been wired up yet, which is being tracked by #7901, but the expected behavior is the service principal would be configured to access that subscription.

Regarding suggested text change, to get us on the same page, by default the command configures the access to all resources under a subscription unless you use `--skip-assignment`. So would the following text be better?
   `Skip creating the default assignment, which allows the service principal to access resources under the current subscription`

Reply:
Got it. This text is clearer.